### PR TITLE
IdSvr: Add prefix to request URI

### PR DESCRIPTION
**Pull request by ngruson**
**What this PR does / why we need it**:
I have several .NET 5 APIs in which I want to check if Identity Server is already reachable. 
My Identity Server is behind an NGINX reverse proxy. The URI looks something like: https://dev.k8s-local.my-app.com/identityserver.
Both the APIs and Identity Server are part of the same Kubernetes deployment.

In described situation, the `GetAsync` call to .well-known/openid-configuration fails with the current version of the IdSvr health check.
With this fix, I can provide a prefix so that the request URI becomes identityserver/.well-known/openid-configuration.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
The new prefix parameter of `AddIdentityServer` is optional.
If you don't use the prefix, code should still compile.

I tried to add unit tests, but my local build of HealthChecks.UI fails.
I just installed NPM version 8.1.4.
```
*** Bundling client files ***
1>
1>> healthchecks.ui@1.0.0 build
1>> npm run build:dll && npm run build:prod
1>
1>
1>> healthchecks.ui@1.0.0 build:dll
1>> webpack-cli --mode production --config ./build/webpack.dll.js
1>
1>node:internal/crypto/hash:67
1>  this[kHandle] = new _Hash(algorithm, xofLen);
1>                  ^
1>
1>Error : error : 0308010C:digital envelope routines::unsupported
1>    at new Hash (node:internal/crypto/hash:67:19)
1>    at Object.createHash (node:crypto:130:10)
1>    at module.exports (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\webpack\lib\util\createHash.js:135:53)
1>    at NormalModule._initBuildHash (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\webpack\lib\NormalModule.js:417:16)
1>    at handleParseError (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\webpack\lib\NormalModule.js:471:10)
1>    at C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\webpack\lib\NormalModule.js:503:5
1>    at C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\webpack\lib\NormalModule.js:358:12
1>    at C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\loader-runner\lib\LoaderRunner.js:373:3
1>    at iterateNormalLoaders (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\loader-runner\lib\LoaderRunner.js:214:10)
1>    at Array.<anonymous> (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\loader-runner\lib\LoaderRunner.js:205:4)
1>    at Storage.finished (C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:55:16)
1>    at C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:91:9
1>    at C:\Repos\ngruson\AspNetCore.Diagnostics.HealthChecks\src\HealthChecks.UI\node_modules\graceful-fs\graceful-fs.js:123:16
1>    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
1>  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
1>  library: 'digital envelope routines',
1>  reason: 'unsupported',
1>  code: 'ERR_OSSL_EVP_UNSUPPORTED'
```

### CODE REVIEW

File: src/HealthChecks.IdSvr/IdSvrHealthCheck.cs [hunk 1]
```diff
@@ -12,16 +12,22 @@ public class IdSvrHealthCheck
...
             _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _prefix = prefix;
         }
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
                 var httpClient = _httpClientFactory();
-                var response = await httpClient.GetAsync(IDSVR_DISCOVER_CONFIGURATION_SEGMENT, cancellationToken);
+                var requestUri = !string.IsNullOrEmpty(_prefix)
```
Comment (sungam3r) [inline src/HealthChecks.IdSvr/IdSvrHealthCheck.cs:26]: Please revert if condition to eliminate `!`.

**Comment by sungam3r** [comment]
Your changes are binary backward incompatible thought they are source level compatible. You may change target branch to `netcore-6.0` so you will be able to test your changes. `netcore-6.0` branch works with configured GitHub Actions for almost each health check project in this repo.

**Comment by sungam3r** [comment]
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/netcore-6.0/.github/workflows/healthchecks_openidconnectserver_ci.yml - CI workflow for IdSvr.

**Comment by ngruson** [comment]
I'll create a new PR based on the netcore-6.0 branch.

**Pull request closed without merging**